src/core/lib/channel/channel_args.cc

```diff
@@ -126,21 +127,53 @@ ChannelArgs ChannelArgs::FromC(const grpc_channel_args* args) {
   return result;
 }
 
-ChannelArgs::CPtr ChannelArgs::ToC() const {
-  std::vector<grpc_arg> c_args;
-  args_.ForEach([&c_args](const std::string& key, const Value& value) {
-    char* name = const_cast<char*>(key.c_str());
-    c_args.push_back(Match(
-        value,
-        [name](int i) { return grpc_channel_arg_integer_create(name, i); },
-        [name](const std::shared_ptr<const std::string>& s) {
-          return grpc_channel_arg_string_create(name,
+grpc_arg ChannelArgs::Value::MakeCArg(const char* name) const {
+  char* c_name = const_cast<char*>(name);
+  return Match(
+        rep_,
+        [c_name](int i) { return grpc_channel_arg_integer_create(c_name, i); },
+        [c_name](const std::shared_ptr<const std::string>& s) {
+          return grpc_channel_arg_string_create(c_name,
                                                 const_cast<char*>(s->c_str()));
```

Comment:
Do we need to worry about `s` being null here?

Reply:
Only if this is moved-from